cache-config/tm-health-client/config/config.go

```diff
@@ -0,0 +1,256 @@
+}
+
+type LogCfg struct {
+	LogLocationErr   string
+	LogLocationDebug string
+	LogLocationInfo  string
+	LogLocationWarn  string
+}
+
+func (lcfg LogCfg) ErrorLog() log.LogLocation   { return log.LogLocation(lcfg.LogLocationErr) }
+func (lcfg LogCfg) WarningLog() log.LogLocation { return log.LogLocation(lcfg.LogLocationWarn) }
+func (lcfg LogCfg) InfoLog() log.LogLocation    { return log.LogLocation(lcfg.LogLocationInfo) }
+func (lcfg LogCfg) DebugLog() log.LogLocation   { return log.LogLocation(lcfg.LogLocationDebug) }
+func (lcfg LogCfg) EventLog() log.LogLocation   { return log.LogLocation(log.LogLocationNull) } // not used
+
+func readCredentials(cfg *Cfg) error {
+	fn := cfg.TOCredentialFile
+	f, err := os.Open(fn)
+
+	if err != nil {
+		return errors.New("failed to open + " + fn + " :" + err.Error())
+	}
+	defer f.Close()
+
+	var to_pass_found = false
+	var to_url_found = false
+	var to_user_found = false
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+		fields := strings.Split(line, " ")
+		for _, v := range fields {
+			if strings.HasPrefix(v, "TO_") {
+				sf := strings.Split(v, "=")
+				if len(sf) == 2 {
+					if sf[0] == "TO_URL" {
+						// parse the url after trimming off any surrounding double quotes
+						cfg.TOUrl = strings.Trim(sf[1], "\"")
+						to_url_found = true
+					}
+					if sf[0] == "TO_USER" {
+						// set the TOUser after trimming off any surrounding quotes.
+						cfg.TOUser = strings.Trim(sf[1], "\"")
+						to_user_found = true
+					}
+					// set the TOPass after trimming off any surrounding quotes.
+					if sf[0] == "TO_PASS" {
+						cfg.TOPass = strings.Trim(sf[1], "\"")
+						to_pass_found = true
+					}
+				}
+			}
+		}
+	}
+	if !to_url_found && !to_user_found && !to_pass_found {
+		return errors.New("failed to retrieve one or more TrafficOps credentails")
+	}
+
+	return nil
+}
+
+func GetConfig() (Cfg, error, bool) {
+	var err error
+	var configFile string
+	var logLocationErr = log.LogLocationStderr
+	var logLocationDebug = log.LogLocationNull
+	var logLocationInfo = log.LogLocationNull
+	var logLocationWarn = log.LogLocationNull
+
+	configFilePtr := getopt.StringLong("config-file", 'f', DefaultConfigFile, "full path to the json config file")
+	logdirPtr := getopt.StringLong("logging-dir", 'l', DefaultLogDirectory, "directory location for log files")
+	helpPtr := getopt.BoolLong("help", 'h', "Print usage information and exit")
+	verbosePtr := getopt.CounterLong("verbose", 'v', `Log verbosity. Logging is output to stderr. By default, errors are logged. To log warnings, pass '-v'. To log info, pass '-vv', debug pass '-vvv'`)
+
+	getopt.Parse()
+
+	if configFilePtr != nil {
+		configFile = *configFilePtr
+	} else {
+		configFile = DefaultConfigFile
+	}
+
+	var logfile string
+
+	logfile = *logdirPtr + "/" + DefaultLogFile
```

Comment:
Nitpick: better to use `filepath.Join` instead of `/`, to work on aberrant operating systems

Reply:
Fixed, I always forget to use filepath.Join() :-)